templates/puppet_signscript.rb.erb

```diff
@@ -0,0 +1,43 @@
+#!/opt/puppetlabs/puppet/bin/ruby
+#
+
+require 'net/http'
+require 'net/https'
+require 'fileutils'
+require 'json'
+require 'yaml'
+
+csr = STDIN.read
+
+settings_file    = File.exist?('/etc/puppetlabs/puppet/foreman.yaml') ? '/etc/puppetlabs/puppet/foreman.yaml' : '/etc/puppet/foreman.yaml'
+SETTINGS         = YAML.load_file(settings_file)
+protocol         = "<%= scope.lookupvar("foreman_proxy::puppetca_listen_on") %>"
```

Comment:
`both` is a valid option in the puppet type. This doesn't handle that currently.